internal/service/ipam/api_dhcp_host_resource_test.go

```diff
@@ -0,0 +1,129 @@
+package ipam_test
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/infobloxopen/bloxone-go-client/ipam"
+
+	"github.com/infobloxopen/terraform-provider-bloxone/internal/acctest"
+)
+
+func TestAccDhcpHostResource_basic(t *testing.T) {
+	var resourceName = "bloxone_dhcp_host.test"
+	var v ipam.Host
+	var dhcpServerName = acctest.RandomNameWithPrefix("dhcp_host")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccDhcpHostBasicConfig("TF_TEST_HOST_02", dhcpServerName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDhcpHostExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttrSet(resourceName, "name"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccDhcpHostResource_Server(t *testing.T) {
+	var resourceName = "bloxone_dhcp_host.test_server"
+	var v ipam.Host
+	var dhcpServerName1 = acctest.RandomNameWithPrefix("dhcp_host")
+	var dhcpServerName2 = acctest.RandomNameWithPrefix("dhcp_host")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccDhcpHostServer("TF_TEST_HOST_02", "server", dhcpServerName1, dhcpServerName2),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDhcpHostExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttrPair(resourceName, "server", "bloxone_dhcp_server.server", "id"),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccDhcpHostServer("TF_TEST_HOST_02", "server2", dhcpServerName1, dhcpServerName2),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDhcpHostExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttrPair(resourceName, "server", "bloxone_dhcp_server.server2", "id"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
```

Comment:
can you test that on delete the configuration is truly removed?

Reply:
Added a CheckDestroy which verifies if the Server object is unassigned from DHCP host.